File: lib-segment/src/main/java/websiteschema/mpsegment/tools/accurary/NerPlaceErrorAnalyzer.java
package websiteschema.mpsegment.tools.accurary;

import websiteschema.mpsegment.core.WordAtom;
import websiteschema.mpsegment.dict.POSUtil;

class NerPlaceErrorAnalyzer extends AbstractErrorAnalyzer {

    @Override
    public boolean analysis(WordAtom expect, String possibleErrorWord) {
        boolean foundError = false;
        if (possibleErrorWord.replaceAll(" ", "").equals(expect.word)) {
            if (expect.pos == POSUtil.POS_NS) {
                increaseOccur();
                addErrorWord(expect.word);
                foundError = true;
            }
        }
        return foundError;
    }
}

File: lib-segment/src/main/scala/websiteschema/mpsegment/tools/accurary/NerPlaceErrorAnalyzer.scala
package websiteschema.mpsegment.tools.accurary

import websiteschema.mpsegment.core.Word
import websiteschema.mpsegment.dict.POSUtil

class NerPlaceErrorAnalyzer extends AbstractErrorAnalyzer {

    override def analysis(expect: Word, possibleErrorWord: String) : Boolean = {
        var foundError = false
        if (possibleErrorWord.replaceAll(" ", "").equals(expect.name)) {
            if (expect.pos == POSUtil.POS_NS) {
                increaseOccur()
                addErrorWord(expect.name)
                foundError = true
            }
        }
        return foundError
    }
}
